v1/server/handlers/compress.go

```diff
@@ -61,17 +61,35 @@ type compressResponseWriter struct {
 	minlength     int
 }
 
-var gzipPool *sync.Pool
+var (
+	gzipPool                 *sync.Pool
+	gzipPoolMutex            sync.RWMutex
+	gzipPoolCompressionLevel int
+)
 
 func initGzipPool(compressionLevel int) {
-	if gzipPool == nil {
-		gzipPool = &sync.Pool{
-			New: func() any {
-				writer, _ := gzip.NewWriterLevel(io.Discard, compressionLevel)
-				return writer
-			},
-		}
+	gzipPoolMutex.RLock()
+	if gzipPool != nil && gzipPoolCompressionLevel == compressionLevel {
+		gzipPoolMutex.RUnlock()
+		return
+	}
+	gzipPoolMutex.RUnlock()
+
+	gzipPoolMutex.Lock()
+	defer gzipPoolMutex.Unlock()
+
+	if gzipPool != nil && gzipPoolCompressionLevel == compressionLevel {
```

Comment:
Can the compression level be reconfigured during runtime?

Reply:
Hmm, perhaps it should be a failure case if it's updated from the original value?

Reply:
Maybe. I think it depends on if the reconfiguration also means that all handlers owning an item from the pool are also dropped/reconfigured. If they do, then all gets subsequent to the reconfig would simply get the updated value, and we should be good. If not, then we might eventually end up with a mix of different active values, which isn't ideal, and we should probably error.